### PR TITLE
Implement Drops

### PR DESCRIPTION
**Pull request by MeikelLP**
TODOs
* [x] Load legacy drop info files
  * [x] `common_drop_item.txt`
  * [x] `etc_drop_item.txt` + from `mob_proto`
  * [x] `mob_drop_item.txt`
  * [x] `drop_item_group.txt` <-- information for this kind of drops is actually contained in `mob_drop_item.txt`
  
* [x] Drop items when killing mobs
* [x] Calculate drop chance based on level
* [x] Ownership
  * [x] Show ownership
  * [x] Enforce ownership 

image

### CODE REVIEW

File: src/Executables/Game/Services/DropProvider.cs [hunk 1]
```diff
@@ -38,12 +45,37 @@ public ImmutableArray<MonsterDropEntry> GetDropsForMob(uint monsterProtoId)
...
+            LoadDeltaPercentagesAsync(cancellationToken),
             LoadDropsForMonstersAsync(cancellationToken),
             LoadSimpleMobDropsAsync(cancellationToken),
             LoadCommonMobDropsAsync(cancellationToken)
         );
     }
 
+    private Task LoadDeltaPercentagesAsync(CancellationToken cancellationToken = default)
+    {
+        _bossPercentageDeltas = _configuration.GetSection("drops:delta:boss")
```
Comment (MeikelLP) [inline src/Executables/Game/Services/DropProvider.cs:57]: You can do:

```cs
_configuration.GetSection("drops:delta:normal").Get<int[]>()
```

File: src/Executables/Game/ParserUtils.cs [hunk 2]
```diff
@@ -249,17 +251,18 @@ private static void ParseCommonDropAndAdd(ReadOnlySpan<char> line, ICollection<C
             read = 0;
             return null;
         }
-
+        
         var minLevel = byte.Parse(line[minLevelStartIndex..minLevelEndIndex]);
         var maxLevel = byte.Parse(line[maxLevelStartIndex..maxLevelEndIndex]);
-        var percentage = float.Parse(line[percentageStartIndex..percentageEndIndex], CultureInfo.InvariantCulture); // TODO check if values are human percentage or math percentage => 50% vs 0.5
+        var percentage = (float) decimal.Parse(line[percentageStartIndex..percentageEndIndex], CultureInfo.InvariantCulture); // math percentage
```
Comment (MeikelLP) [inline src/Executables/Game/ParserUtils.cs:257]: Why parse as decimal and cast to float?

**Comment by MeikelLP** [comment]
@WoozChucky is this done? If yes I can merge it for u (overriding sonar rule)

**Comment by WoozChucky** [comment]
I think at this stage, we either need to implement skills, quests and riding to implement these specific drops differently, or merge it like this and when we implement said systems, it needs to be coded in the drops logic.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by MeikelLP** [comment]
@WoozChucky I think it's fine in the current state. More features will be added as dependencies get implemented. Feel free to write these dependencies down somehow (roadmap?)

**Pull request merged**